src/server/node_services/nodes_monitor.js

```diff
@@ -1267,121 +1267,154 @@ class NodesMonitor extends EventEmitter {
-                connection: item.connection
-            })
-        );
+        try {
+            await P.timeout(AGENT_RESPONSE_TIMEOUT,
+                this.client.agent.test_store_validity(null, {
+                    connection: item.connection
+                })
+            );
+        } catch (err) {
+            // ignore "unkonown" errors - we don't want to put the node in detention in cases where we don't know what is the problem
+            // if there is a real issue, we will take it into account in report_error_on_node_blocks
+            if (err.rpc_code !== 'AUTH_FAILED' && err.rpc_code !== 'STORAGE_NOT_EXIST') {
+                dbg.warn(`encountered an unknown error in _test_store_perf`, err);
+            } else {
+                throw err;
+            }
+
+        }
     }
 
     async _test_store_perf(item) {
         const now = Date.now();
         if (item.last_store_perf_test && now < item.last_store_perf_test + STORE_PERF_TEST_INTERVAL) return;
+        try {
 
 
-        dbg.log1('running _test_store_perf::', item.node.name);
-        const res = await P.timeout(AGENT_RESPONSE_TIMEOUT,
-            this.client.agent.test_store_perf({
-                count: 5
-            }, {
-                connection: item.connection
-            })
-        );
-        item.last_store_perf_test = Date.now();
-        dbg.log0(`_test_store_perf for node ${item.node.name} returned:`, res);
-        this._set_need_update.add(item);
-        item.node.latency_of_disk_read = js_utils.array_push_keep_latest(
-            item.node.latency_of_disk_read, res.read, MAX_NUM_LATENCIES);
-        item.node.latency_of_disk_write = js_utils.array_push_keep_latest(
-            item.node.latency_of_disk_write, res.write, MAX_NUM_LATENCIES);
+            dbg.log1('running _test_store_perf::', item.node.name);
+            const res = await P.timeout(AGENT_RESPONSE_TIMEOUT,
+                this.client.agent.test_store_perf({
+                    count: 5
+                }, {
+                    connection: item.connection
+                })
+            );
+            item.last_store_perf_test = Date.now();
+            dbg.log0(`_test_store_perf for node ${item.node.name} returned:`, res);
+            this._set_need_update.add(item);
+            item.node.latency_of_disk_read = js_utils.array_push_keep_latest(
+                item.node.latency_of_disk_read, res.read, MAX_NUM_LATENCIES);
+            item.node.latency_of_disk_write = js_utils.array_push_keep_latest(
+                item.node.latency_of_disk_write, res.write, MAX_NUM_LATENCIES);
+        } catch (err) {
+            // ignore "unkonown" errors - we don't want to put the node in detention in cases where we don't know what is the problem
+            // if there is a real issue, we will take it into account in report_error_on_node_blocks
+            if (err.rpc_code !== 'AUTH_FAILED' && err.rpc_code !== 'STORAGE_NOT_EXIST') {
```

Comment:
As these errors are very specific to cloud I think we should maybe check this only for cloud and handle differently pv pool errors.